bindsnet/pipeline/environment_pipeline.py

```diff
@@ -79,42 +74,38 @@ def __init__(
         self.reward_plot = None
 
         self.first = True
-        self.analyzer = MatplotlibAnalyzer()
+        self.analyzer = MatplotlibAnalyzer(**self.plot_config)
 
-    def init_fn(self):
+    def init_fn(self) -> None:
         pass
 
-    def train(self) -> None:
+    def train(self, **kwargs) -> None:
+        # language=rst
         """
-        Runs for the specified number of episodes from the Environment.
-        Each episode can be an arbitrary length.
+        Runs for the specified number of episodes. Each episode can be an arbitrary length.
         """
-
-        for self.episode in range(self.num_episodes):
+        while self.episode < self.num_episodes:
```

Comment:
This is unnecessary, whats wrong with for?

Reply:
The `for` loop doesn't allow you to use the `self.episode` anywhere else, so there would be no point in making it an attribute of `self`. Besides, I made use of `self.episode` in `EnvironmentPipeline.plots()`, in order to take care of plotting reward only so many episodes.